### ISSUE TITLE
Valid Loss不下降的问题

### ISSUE THREAD

**Issue by soya-git** [SJTU-Quant/MASTER#13]
![loss](https://github.com/SJTU-Quant/MASTER/assets/126449058/05a0ec5b-d0eb-4f7f-bbee-94e5ade167c1)
您好！请问根据您提供的数据和源码，为何从开始训练时，虽然train loss不断下降，但valid loss持续震荡最终并未收敛。请问您在工作中遇到了这种情况吗？您觉得这是什么原因造成的？

**Comment by soya-git** [comment]
Actually I noticed someone has asked this question. According to the answer before, it is an overfitting problem caused by incorrect parameters. However, we did not change any parameter setting in the code and it seems that the overfitting starts from the 6 epoch, which is quite early.  I am confused of how this could happen.

**Comment by LITONG99** [comment]
For reproduction, please set the random seed as 1, 2, 3, 4, and 5 and run the code.  For this specific run, the trend of validation loss is reasonable.

The loss changes in parallel only if the validation set distribution and the training set distribution are identical, which is rarely the case in the stock domain. As we split the training, validation, and test set chronologically, there can be some gap between the distributions.  So, yes, the validation loss will oscillate and probably not converge. The phenomenon can be observed if one experiments with other models and we believe it is an inherent feature of the stock prediction task. 

It also implies that the sole validation loss would not provide a good reference to the test performance, and in our study, both training and validation loss, and repeated experiments are considered to end the model training, given that they include all the prior data we know before the test. 

I would like to correct that overfitting is not caused by incorrect parameters, in practice, it nearly always happens if one keeps optimizing the models by training loss.

**Comment by soya-git** [comment]
Thank you for your patient response.  Your explanation helps  a lot.

**Comment by soya-git** [comment]
Sorry for the trouble caused, but after trying seed from 0-20, the valid loss problem exists consistently.
